noxfile.py

```diff
@@ -117,8 +119,14 @@ def set_dev(session):
     """Install PyBaMM in editable mode."""
     set_environment_variables(PYBAMM_ENV, session=session)
     envbindir = session.bin
-    session.install("-e", ".[all]", silent=False)
+    session.install("virtualenv")
+    session.run("virtualenv", os.fsdecode(VENV_DIR), silent=True)
+    python = os.fsdecode(VENV_DIR.joinpath("bin/python"))
+    session.run(python, "-m", "pip", "install", "-e", ".[all]", silent=False)
     session.install("cmake", silent=False)
+    session.install("-e", ".[dev]", external=True)
+    if session.platform.startswith("macos") or session.platform.startswith("linux"):
```

Comment:
Could you use `sys.platform` here instead, just to be consistent with how the operating system is detected at other locations in the file?